### ISSUE TITLE
Missing SPDX identifier data

### ISSUE THREAD

**Issue by goneall** [OpenSourceOrg/licenses#62]
The following SPDX identifiers are not in this repo/JSON file but are listed on the [OSI website](https://opensource.org/licenses/alphabetical):
- 0BSD
- BSD-2-Clause-Patent
- BSD-1-Clause
- BSD-3-Clause-LBNL
- CAL-1.0
- EPL-2.0
- EUPL-1.2
- CECILL-2.1
- OLDAP-2.8
- PHP-3.01
- MIT-0

The SPDX license ID on the website is incorrect for the following:
- Licence Libre du Québec – Réciprocité forte (LiLiQ-R+) version 1.1 (LiliQ-R+) - should be (LiLiQ-Rplus-1.1)
- Licence Libre du Québec – Réciprocité (LiLiQ-R) version 1.1 (LiliQ-R) - should be (LiLiQ-R-1.1)
- Licence Libre du Québec – Permissive (LiLiQ-P) version 1.1 (LiliQ-P) - should be (LiLiQ-P-1.1)
- Mulan Permissive Software License v2 (MulanPSL - 2.0) - just a typo - no space before or after the dash
- Universal Permissive License (UPL) - should be (UPL-1.0)

The following licenses are listed by name on the [OSI website](https://opensource.org/licenses/alphabetical) but are missing the SPDX identifiers:
- CERN Open Hardware Licence Version 2 - Permissive (CERN-OHL-P-2.0)
- CERN Open Hardware Licence Version 2 - Weakly Reciprocal (CERN-OHL-S-2.0)
- CERN Open Hardware Licence Version 2 - Strongly Reciprocal (CERN-OHL-W-2.0)
- OSET Public License version 2.1 (OSET-PL-2.1)
- Upstream Compatibility License v1.0 (UCL-1.0)
- The Unlicense (Unlicense)
- Unicode Data Files and Software License (Unicode-DFS-2016)

Although not an inconsistency between the website and this file, I would also like to add some additional SPDX identifiers to the GPL family of licenses.  Per request from the Free Software Foundation a couple years ago, we added license identifiers with a suffix "-or-later" to the GPL and LGPL family of licenses.  If would be nice if we added those identifiers to this JSON representation.  It may also make sense to add them to the website.

**Comment by webmink** [comment]
Anticipating some more pull requests for the remaining items raised, which I'll add as they arise.

**Comment by goneall** [comment]
@webmink I added PR #69 

I'm wondering if multiple PR's are going to create merge conflicts.  I'm going to wait until 0BSD is merged before adding any more PR's in case the current process causes conflicts.

**Comment by webmink** [comment]
OK. Hopefully we'll be caught up in the next few days!

**Comment by goneall** [comment]
I just put in PR's for all remaining SPDX identifiers are not in this repo/JSON file but are listed on the OSI website.

Note CECILL-2.1 was already there - so no need for a PR.

Once these are merged, I'll create PRs for the remaining issues.

**Comment by webmink** [comment]
Since licensing matters are ultimately overseen for OSI by @pchestek I'm adding her to all of them as an approver. Once she's given us the nod either I or @paultag will merge them.

**Comment by goneall** [comment]
@paultag @webmink Several of the licenses present on the website with missing SPDX id's are not present in this repo.

Would you like me to create a PR which adds the licenses along with the SPDX ID's missing from the website or did you want to sync up this repo with the website content?

Below is the list of licenses I found on the website without corresponding text in this repo:

 CERN Open Hardware Licence Version 2 - Permissive (CERN-OHL-P-2.0)
 CERN Open Hardware Licence Version 2 - Weakly Reciprocal (CERN-OHL-S-2.0)
 CERN Open Hardware Licence Version 2 - Strongly Reciprocal (CERN-OHL-W-2.0)
 Mulan Permissive Software License v2 (MulanPSL-2.0)
 Upstream Compatibility License v1.0 (UCL-1.0)
 The Unlicense (Unlicense)
 Unicode Data Files and Software License (Unicode-DFS-2016)

**Comment by richardfontana** [comment]
> The SPDX license ID on the website is incorrect for the following:

@goneall in general it is incorrect to say these are errors. I am responsible for a few of those cases as I was on the OSI board at the time of the license approval.  The case I remember pretty clearly is that of the Licence Libre du Québec licenses. The abbreviations you mention are the ones used by the license steward in the course of making their license submission and (IIRC) reflected in the license texts themselves. At the time, we weren't trying to root out all possible forms of referring to OSI-approved licenses in abbreviated form other than using SPDX identifiers. It's sort of like saying any mention of "GPLv2" is incorrect because "GPLv2" is not an SPDX identifier. Different or alternative system of license abbreviation != "incorrect".  It's possible that since the time I left the OSI board the OSI has chosen to more ambitiously or consistently promote use of SPDX identifiers.

**Comment by goneall** [comment]
> @goneall in general it is incorrect to say these are errors. 

@richardfontana I'm not commenting on the OSI ID's, I'm specifically referring to the SPDX identifiers on the OSI website.

On the [License Listed Alphabetically](https://opensource.org/licenses/alphabetical), the following description of the list is provided:

> The following licenses have been approved by the OSI. The parenthesized expression following a license name is its SPDX short identifier (if one exists).

Looking at the SPDX Id's specifically, I found the inconsistency with the SPDX ID's on the [SPDX listed license](https://spdx.org/licenses/) pages.

For example, the following appears on the OSI page:
`Licence Libre du Québec – Réciprocité forte (LiLiQ-R+) version 1.1 (LiliQ-R+)`

while the SPDX license list has:
`Licence Libre du Québec – Réciprocité forte version 1.1 | LiLiQ-Rplus-1.1`

Since the OSI web page specifically mentions the SPDX ID for the ID in parenthesis following the name, I would expect the ID to match the SPDX listed license ID.  Let me know if you disagree.

**Comment by richardfontana** [comment]
> Since the OSI web page specifically mentions the SPDX ID for the ID in parenthesis following the name, I would expect the ID to match the SPDX listed license ID. Let me know if you disagree.

@goneall you are correct, I misunderstood. Given that you can see the care with which BSD+Patent is handled on this list (with "BSD-2-Clause-Patent" as the parenthesized identifier) I can't explain what happened with the other cases.

**Comment by kopeboy** [comment]
Any update on this?! I was trying to import the list of OSI-approved licenses to my platform but the link for the JSON in the readme is not consistent with the info on OSI website.. this is a critical bug imho.
I liked the "keywords" you provide to be able to filter, but I guess importing from SPDX is much safer since their repository seems maintained and api + website implementations are consistent.

**Comment by goneall** [comment]
@kopeboy - I was waiting for PR #82 to be merged before making any additional changes.

@webmink - Any progress on the PR?

**Comment by webmink** [comment]
Done.

**Comment by goneall** [comment]
Thanks @webmink

@kopeboy - I have some major release for some SPDX tools this week, but I'll try to get to the other PR's.

BTW - feel free to create PR's for these yourself if you'd like to move these along more quickly - we're all just volunteers ;)
